docs/proposals/13-automated-seed-management.md

```diff
@@ -263,18 +283,66 @@ ManagedSeedSets are reconciled by a new *managed seed set controller* in `garden
+
+There is an important difference between seeds and pods or nodes in that seeds are more "heavyweight" and therefore updating a set of seeds by introducing new seeds and moving shoots to them tends to be much more complex, time-consuming, and prone to failures compared to updating the seeds "in place". Furthermore, updating seeds in this way depends on a mature implementation of [GEP-7: Shoot Control Plane Migration](07-shoot-control-plane-migration.md), which is not available right now. Due to these considerations, we favor the second approach over the first one.
+
+#### ManagedSeed Identity and Order
+
+A StatefulSet manages the deployment and scaling of a set of Pods, and provides guarantees about the ordering and uniqueness of these Pods. It maintains a *stable identity* (including network identity) for each of their Pods. These pods are created from the same spec, but are not interchangeable: each has a persistent identifier that it maintains across any rescheduling.
+
+A StatefulSet achieves the above by associating each replica with an *ordinal number*. With n replicas, these ordinal numbers range from 0 to n-1. When scaling out, newly added replicas always have ordinal numbers larger than those of previously existing replicas. When scaling in, it is the replicas with the largest orginal numbers that are removed.
+
+Besides stable identity and persistent storage, these ordinal numbers are also used to implement the following StatefulSet features: 
+
+* Ordered, graceful deployment and scaling.
+* Ordered, automated rolling updates. Such rolling updates can be *partitioned* (limited to replicas with ordinal numbers greater than or equal to the "partition") to achieve *phased rollouts*.
+
+A ManagedSeedSet, unlike a StatefulSet, does not need to maintain a stable identity for its ManagedSeeds. Furthermore, it would not be practical to always remove the replicas with the largest ordinal numbers when scaling in, since the corresponding seeds may have shoots scheduled onto them, while other seeds, with lower ordinals, may have fewer shoots (or none), and therefore be much better candidates for being removed.
+
+On the other hand, it would be beneficial if a ManagedSeedSet, like a StatefulSet, provides ordered deployment and scaling, ordered rolling updates, and phased rollouts. The main advantage of these features is that a deployment or update failure would affect fewer replicas (ideally just one), containing any potential damage and making the situation easier to handle, thus achieving some of the goals stated in [Issue #87](https://github.com/gardener/gardener/issues/87).
```

Comment:
I can also imagine that a phased rollout would help to contain Seed rolling updates outside business hours to avoid Seed being rolled when most stakeholders are wanting to use their clusters.

Reply:
OK, I will mention this as well, thanks.